binding_generator.py

```diff
@@ -79,20 +99,31 @@ def generate_class_header(used_classes, c):
     
     source.append("namespace godot {")
     source.append("")
-    
-    
+
+
     for used_type in used_classes:
-        source.append("class " + strip_name(used_type) + ";")
+        if is_enum(used_type) or is_nested_type(used_type, class_name):
+            continue
+        else:
+            source.append("class " + strip_name(used_type) + ";")
     
     
     source.append("")
     
     
     # generate the class definition here
-    source.append("class " + strip_name(c["name"]) + ("" if c["base_class"] == "" else (" : public " + strip_name(c["base_class"])) ) + " {")
+    source.append("class " + class_name + ("" if c["base_class"] == "" else (" : public " + strip_name(c["base_class"])) ) + " {")
     
     source.append("public:")
     source.append("")
+
+    for enum in c["enums"]:
+        source.append("\tenum " + strip_name(enum["name"]) + " {")
+        # for value in enum["values"]:
+        #     source.append("\t\t" + value + " = " + str(enum["values"][value]) + ",")
+        source.append("\t};")
```

Comment:
Shouldn't this be uncommented again?

Reply:
Now it produces compilation fail, because classes already have constants with similar names. I could add checking and if constant already exist in enum, don't write it as constant to file.